deps/chakrashim/core/lib/Jsrt/Jsrt.cpp

```diff
@@ -3433,6 +3433,178 @@ CHAKRA_API JsStringFree(_In_ char* stringValue)
     return JsNoError;
 }
 
+
+// -------- Experimental new String APIs ----------------------------------
+
+
+template <class SrcChar, class DstChar>
+static void CastCopy(const SrcChar* src, DstChar* dst, size_t count)
+{
+    const SrcChar* end = src + count;
+    while (src < end)
+    {
+        *dst++ = static_cast<DstChar>(*src++);
```

Comment:
Curious, what happens when `sizeof(SrcChar) > sizeof(DstChar)`? I see below that you are using this method where src is char16 and dst is char. Don't we miss out 8 bits in casting?